pkg/monitoring/metrics/virt-controller/vmlabels_config.go

```diff
@@ -0,0 +1,178 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package virt_controller
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	k6tv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+	clientutil "kubevirt.io/client-go/util"
+)
+
+// LabelsConfig provides an interface for loading and evaluating VM label configuration.
+// Load should refresh the configuration from its source, and ShouldReport determines whether
+// a specific label key should be exposed given the current allow/deny lists.
+// NewLabelsConfig constructs a LabelsConfig backed by a Kubernetes ConfigMap.
+
+type LabelsConfig interface {
+	Load() error
+	ShouldReport(vm *k6tv1.VirtualMachine, label string) bool
+}
+
+func NewLabelsConfig() LabelsConfig {
+	return &labelsConfigImpl{}
+}
+
+var (
+	defaultVMLabelsAllowlist  = []string{"*"}
+	defaultVMLabelsIgnorelist = []string{}
+
+	// Current allow/ignore lists for VM labels
+	vmLabelsAllowlist  = defaultVMLabelsAllowlist
+	vmLabelsIgnorelist = defaultVMLabelsIgnorelist
+
+	// Protects updates and reads of vmLabelsAllowlist/vmLabelsIgnorelist
+	vmLabelsConfigMu sync.RWMutex
+
+	// ConfigMap name for VM labels configuration
+	vmLabelsConfigMapName = "kubevirt-vm-labels-config"
+)
+
+// labelsConfigImpl is a ConfigMap-backed implementation of LabelsConfig.
+type labelsConfigImpl struct{}
+
+func (c *labelsConfigImpl) Load() error {
+	loadVMLabelsConfiguration()
+	return nil
+}
+
+func (c *labelsConfigImpl) ShouldReport(vm *k6tv1.VirtualMachine, label string) bool {
+	vmLabelsConfigMu.RLock()
+	allowlist := vmLabelsAllowlist
+	ignorelist := vmLabelsIgnorelist
+	vmLabelsConfigMu.RUnlock()
+
+	if len(allowlist) == 0 {
+		return false
+	}
+
+	allowAll := false
+	allowMap := make(map[string]bool)
```

Comment:
doesnt make sense the use of the map here, you are not saving it for later use
you always recreate it when you call ShouldReport for a specific label

Load() should not be in the interface in my opinion
you can call `loadVMLabelsConfiguration` once on `NewLabelsConfig`
but then have an event watcher to trigger on updates `informer.AddEventHandler....`

Reply:
fixed